### PR TITLE
bugfix/NLRFIM-76 - NotOnOrAfter validation

### PR DESCRIPTION
**Pull request by Snede**
Suggested fix for Issue #33.

This PR contains:
1. Fixing issue where AllowedClockSkewMinutes would narrow the valid window rather than extending it.
2. Extracting the validation logic into its own method.
3. Added missing NotOnOrAfter validation to GET binding.
4. Added missing NotOnOrAfter validation to SOAP binding.
5. Fixed minor spelling mistake.

### CODE REVIEW

File: src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs [hunk 1]
```diff
@@ -621,6 +613,31 @@ private void HandleRequest(HttpContext context)
...
+                return;
+            
+            var notOnOrAfter = logoutRequest.NotOnOrAfter.Value;
+            var allowedClockSkewTime = FederationConfig.GetConfig().AllowedClockSkewMinutes;
+            var now = DateTime.UtcNow;
+            
+            if (notOnOrAfter.AddMinutes(allowedClockSkewTime) > now) 
+                return;
+            
+            var errormessage = $"Logout request expired. NotOnOrAfter={notOnOrAfter}, RequestReceived={allowedClockSkewTime}";
```
Comment (sihpe) [inline src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs:637]: Skal det ikke være `RequestReceived={now}` ?

File: src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs [hunk 2]
```diff
@@ -621,6 +613,31 @@ private void HandleRequest(HttpContext context)
...
+        /// Validates that the <see cref="LogoutRequest"/> has not expired, by checking that NotOnOrAfter + allowedTimeSkew is greater than current time.
+        /// Only validated if <see cref="LogoutRequest.NotOnOrAfter"/> is not null.
+        /// </summary>
+        /// <param name="context">The requests <see cref="HttpContext"/></param>
+        /// <param name="logoutRequest">The LogoutRequest to validate</param>
+        /// <exception cref="T:System.Threading.ThreadAbortException">Throws ThreadAbortException if validation fails, and config is set to show an error page.</exception>
+        /// <exception cref="Saml20Exception">Throws Saml20Exception if validation fails, and config is set to throw exception.</exception>
+        private void ValidateNotOnOrAfter(HttpContext context, LogoutRequest logoutRequest)
+        {
+            if (!logoutRequest.NotOnOrAfter.HasValue)
```
Comment (sihpe) [inline src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs:627]: Jeg kan bedst lide ifs med curly-braces ... (gælder også næste if)

File: src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs (hunk 1 above)
Comment (Snede) [reply src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs]: Rettet.

File: src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs (hunk 2 above)
Comment (Snede) [reply src/dk.nita.saml20/dk.nita.saml20/Protocol/Saml20LogoutHandler.cs]: Rettet.

**Review by sihpe — APPROVED** [review]

**Pull request merged**